### PR TITLE
[build] Cleanup dependency of distributed flags

### PR DESCRIPTION
**Pull request by orionr**
Now that we're building everything together, making all distributed flags conditional of USE_DISTRIBUTED being set.

cc @pietern @teng-li @cpuhrsch

### CODE REVIEW

File: cmake/Summary.cmake [hunk 1]
```diff
@@ -125,7 +120,10 @@ function (caffe2_print_configuration_summary)
   message(STATUS "  USE_ZMQ               : ${USE_ZMQ}")
   message(STATUS "  USE_DISTRIBUTED       : ${USE_DISTRIBUTED}")
   if(${USE_DISTRIBUTED})
-    message(STATUS "    USE_DISTRIBUTED_MW     : ${USE_DISTRIBUTED_MW}")
+    message(STATUS "    USE_DISTRIBUTED_MW  : ${USE_DISTRIBUTED_MW}")
```
Comment (pietern) [inline cmake/Summary.cmake]: I think `DISTRIBUTED_MW` can go. This has been disabled for a while and I doubt it still compiles. If it hasn't been removed from the repo altogether yet.

Comment (orionr) [reply cmake/Summary.cmake]: Should this be removed as well? https://github.com/pytorch/pytorch/blob/master/torch/csrc/Module.cpp#L602-L612

Comment (pietern) [reply cmake/Summary.cmake]: Yes, good catch! See #10731 for corresponding PR that removed the master worker stuff.

**Comment by teng-li** [comment]
Oh BTW USE_MPI will not affect THD and C10D currently.

File: cmake/Summary.cmake (hunk 1 above)
Comment (pietern) [reply cmake/Summary.cmake]: @orionr See #11240

**Comment by orionr** [comment]
@teng-li in that case, should I keep it separate? As in, not under the `USE_DISTRIBUTED` check? Let me know.

**Comment by teng-li** [comment]
@orionr  the thing is THD and C10D's cmakefile will not check this flags, maybe we should let them check USE_MPI. And only detect MPI and build if USE_MPI is set?

**Pull request closed without merging**